### ISSUE TITLE
Signal Plot: unable to draw the lines with LineStyle.Dash

### ISSUE THREAD

**Issue by damiandixon** [ScottPlot/ScottPlot#929]
Using master branch.

Changed the cookbook/Plottable/Signal.cs file as follows:
```cs
        public void ExecuteRecipe(Plot plt)
        {
            double[] ys = DataGen.RandomWalk(null, 500);
            int sampleRate = 10;

            var sp2 = plt.AddSignal(ys, sampleRate, Color.Magenta);
            sp2.OffsetY = 1000;
            sp2.OffsetX = 300;
            sp2.LineStyle = LineStyle.Dash;
            sp2.LineWidth = 4;
        }
```
I set the LineStyle and LineWidth. The LineWidth mainly to convince myself I was looking at the correct example.

I see the following when I display Plottable: Signal Plot->Styled Signal Plot:

![image](https://user-images.githubusercontent.com/1237723/113137503-8602c480-921c-11eb-9004-17cbd66d4015.png)

**Comment by damiandixon** [comment]
The problem is I believe in SignalPlotBase.cs as follows:

```
        public virtual void Render(PlotDimensions dims, Bitmap bmp, bool lowQuality = false)
        {
            using var gfx = GDI.Graphics(bmp, dims, lowQuality);
            using var brush = GDI.Brush(Color);
            using var penLD = GDI.Pen(Color, (float)LineWidth, LineStyle, true);
            using var penHD = GDI.Pen(Color, (float)LineWidth, LineStyle.Solid, true);
```
The last line of the above section I believe should be:
```
            using var penHD = GDI.Pen(Color, (float)LineWidth, LineStyle, true);
```
Which gives me:

![image](https://user-images.githubusercontent.com/1237723/113138230-6b7d1b00-921d-11eb-915f-dcb699edc8ba.png)

**Comment by damiandixon** [comment]
There is also a similar issue with the SignalPlotXYGeneric.cs:

I think the code should look like:
```
        public override void Render(PlotDimensions dims, Bitmap bmp, bool lowQuality = false)
        {
            using (Graphics gfx = GDI.Graphics(bmp, dims, lowQuality))
            using (var brush = new SolidBrush(Color))
            using (var penHD = GDI.Pen(Color, (float)LineWidth, LineStyle, true))
```

**Comment by StendProg** [comment]
Hi @damiandixon,
This is by design, there is no mistake.

The `Signals` uses a trick that allows you not to draw millions of lines at a time, but at the same time have the correct visual result, as if these million lines were drawn. This trick actually works absolutely correctly only with solid lines that are 1 pixel thick. It is not possible to generalize this trick to make it work for different dash line styles.
If you draw a scatter with a million points with a dashed line, then oddly enough you will see a solid line when you look at the entire chart at once. So, to some extent, the current behavior of the `Signals` is correct. The largest discrepancies will be some average density (say 5000 points on the screen), when the dashed lines only partially overlap, but I'm afraid that nothing can be done about it. At low densities, the current implementation also uses the correct dash line style.

All in all, I would just advise against using dash lines for drawing `Signals`. and distinguish them by color. Or continue to use given the restrictions that I described above.

**Comment by StendProg** [comment]
All the same, you are right and there is a mistake, at low density `penLD` should be used, but `penHD` is used.
This line:
https://github.com/ScottPlot/ScottPlot/blob/822b975be9d4b5f5d1b5057782bcc4f6e0d7740b/src/ScottPlot/Plottable/SignalPlotBase.cs#L279
should be changed to:
```c#
gfx.DrawLines(penLD, pointsArray);
```
I will make PR soon, but my recommendation to use solid lines for `Signals` is still valid.

**Comment by swharden** [comment]
Thanks @damiandixon for reporting this, and @StendProg for the quick fix! I merged #931 and will release an updated package later today 👍
